src/commonMain/kotlin/exchange.dydx.abacus/processor/router/skip/SkipProcessor.kt

```diff
@@ -85,19 +85,45 @@ internal class SkipProcessor(
         payload: Map<String, Any>,
         requestId: String?,
     ): Map<String, Any>? {
-        throw NotImplementedError("receivedRoute is not implemented in SkipProcessor!")
+        var modified = mutableMapOf<String, Any>()
+        existing?.let {
+            modified = it.mutable()
+        }
+        val tokenAddress = parser.asString(parser.value(modified, "transfer.token"))
+        val selectedChainId = parser.asString(parser.value(modified, "transfer.chain"))
+        val decimals = parser.asDouble(selectedTokenDecimals(tokenAddress = tokenAddress, selectedChainId = selectedChainId))
+        val processor = SkipRouteProcessor(parser)
+
+        modified.safeSet(
+            "transfer.route",
+            processor.received(null, payload, decimals = decimals) as MutableMap<String, Any>,
+        )
+        if (requestId != null) {
```

Comment:
Does skip api have requestId concept as well?

Reply:
they do send back a `Grpc-Metadata-Traceid` which we can use as a unique identifier for the request. 

that being said - i don't think we need that param anymore. i think it was only used in the `fetchSquidStatus` method in web as a header. the skip API won't require us to store a trace/request id

Reply:
ah i just realized, there was an opportunity to clean this up. will do in a future PR 🫡